apps/emqx_dashboard/src/emqx_dashboard_api.erl

```diff
@@ -211,8 +213,14 @@ user(delete, #{bindings := #{username := Username}}) ->
         true -> {400, #{code => <<"CONNOT_DELETE_ADMIN">>,
                         message => <<"Cannot delete admin">>}};
         false ->
-            _ = emqx_dashboard_admin:remove_user(Username),
-            {200}
+            case emqx_dashboard_admin:lookup_user(Username) of
+                [] ->
+                    Message = list_to_binary(io_lib:format("Username [~p] not exist", [Username])),
+                    {404, 'NOT_FOUND', Message};
+                _ ->
+                    _ = emqx_dashboard_admin:remove_user(Username),
```

Comment:
this is done in a transaction:
```
-spec(remove_user(binary()) -> ok | {error, any()}).
remove_user(Username) when is_binary(Username) ->
    Trans = fun() ->
                    case lookup_user(Username) of
                    [] ->
                        mnesia:abort(<<"Username Not Found">>);
                    _  -> ok
                    end,
                    mnesia:delete({mqtt_admin, Username})
            end,
    return(ekka_mnesia:transaction(?DASHBOARD_SHARD, Trans)).
```
please make the transaction return `{error, not_found}`

Reply:
Changed